<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from src/manipulator/Cobod_RoBetArme_ROS2_Description/urdf/robetarme.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="robetarme">
  <!-- <xacro:property name="op_mode" default="${int(arg('op_mode'))}"/> -->
  <!-- create link fixed to the "world" -->
  <link name="world"/>
  <ros2_control name="robetarme" type="system">
    <hardware>
      <plugin>mock_components/GenericSystem</plugin>
      <param name="mock_sensor_commands">false</param>
      <param name="mock_gpio_commands">false</param>
      <param name="state_following_offset">0.0</param>
      <param name="calculate_dynamics">true</param>
    </hardware>
    <joint name="Joint_0">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_1">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">1.047</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_2">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">-1.047</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_3">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_4">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_5">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_6">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
  </ros2_control>
  <link name="base_link"/>
  <link name="base_link_inertia">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/base_link.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/base_link.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.2901"/>
      <mass value="135"/>
      <inertia ixx="5.9352" ixy="0.0" ixz="0.0" iyy="4.89026" iyz="0.0" izz="9.19377"/>
    </inertial>
  </link>
  <link name="Link_1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/Link_1.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/Link_1.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00231 -0.00551 0.29945"/>
      <mass value="160"/>
      <inertia ixx="5.00391" ixy="0.0" ixz="0.0" iyy="4.09141" iyz="0.0" izz="5.62542"/>
    </inertial>
  </link>
  <link name="Link_2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/Link_2.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/Link_2.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.59655 -0.01645 0.0"/>
      <mass value="120"/>
      <inertia ixx="3.83235" ixy="0.0" ixz="0.0" iyy="2.20509" iyz="0.0" izz="4.14067"/>
    </inertial>
  </link>
  <link name="Link_3">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/Link_3.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/Link_3.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.2594 -0.00987 0.2143"/>
      <mass value="106"/>
      <inertia ixx="1.81729" ixy="0.0" ixz="0.0" iyy="1.58749" iyz="0.0" izz="2.67974"/>
    </inertial>
  </link>
  <link name="Link_4">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/Link_4.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/Link_4.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.15642 0.0 0.00827"/>
      <mass value="32.7"/>
      <inertia ixx="0.28664" ixy="0.0" ixz="0.0" iyy="0.27189" iyz="0.0" izz="0.11751"/>
    </inertial>
  </link>
  <link name="Link_5">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/Link_5.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/Link_5.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.24971 -0.00243 -0.0047"/>
      <mass value="33.5"/>
      <inertia ixx="0.4212" ixy="0.0" ixz="0.0" iyy="0.29617" iyz="0.0" izz="0.33629"/>
    </inertial>
  </link>
  <link name="Link_6">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/Link_6.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/Link_6.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.22638 0.00445 0.00428"/>
      <mass value="18"/>
      <inertia ixx="0.08556" ixy="0.0" ixz="0.0" iyy="0.06232" iyz="0.0" izz="0.08946"/>
    </inertial>
  </link>
  <link name="Scraper_Link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/visual/Scraper_Link.dae" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robetarme_description/meshes/cobod_arm/collision/Scraper_Link.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <mass value="0.5"/>
      <inertia ixx="0.00023" ixy="0.0" ixz="0.0" iyy="0.00013" iyz="0.0" izz="0.00011"/>
    </inertial>
  </link>
  <joint name="base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="base_link"/>
  </joint>
  <joint name="base_link-base_link_inertia" type="fixed">
    <parent link="base_link"/>
    <child link="base_link_inertia"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="Joint_0" type="revolute">
    <parent link="base_link_inertia"/>
    <child link="Link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.5191"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="2.0943951023931953"/>
    <dynamics damping="1.0" friction="100.0"/>
  </joint>
  <joint name="Joint_1" type="revolute">
    <parent link="Link_1"/>
    <child link="Link_2"/>
    <origin rpy="0 0 0" xyz="0 0 0.3665"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="0.0" upper="2.2689280275926285" velocity="2.0943951023931953"/>
    <dynamics damping="1.0" friction="100.0"/>
  </joint>
  <joint name="Joint_2" type="revolute">
    <parent link="Link_2"/>
    <child link="Link_3"/>
    <origin rpy="0 0 0" xyz="-1.047 0.0095911 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="-0.12217304763960307" velocity="2.6179938779914944"/>
    <dynamics damping="1.0" friction="100.0"/>
  </joint>
  <joint name="Joint_3" type="revolute">
    <parent link="Link_3"/>
    <child link="Link_4"/>
    <origin rpy="0 0 0" xyz="0.79956 -0.010459 0.27902"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.6651914291880923"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <joint name="Joint_4" type="revolute">
    <parent link="Link_4"/>
    <child link="Link_5"/>
    <origin rpy="0 0 0" xyz="0.25836 -0.0023493 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.6651914291880923"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <joint name="Joint_5" type="revolute">
    <parent link="Link_5"/>
    <child link="Link_6"/>
    <origin rpy="0 0 0" xyz="0.381 0 0.0024824"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.6651914291880923"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="Joint_6" type="revolute">
    <parent link="Link_6"/>
    <child link="Scraper_Link"/>
    <origin rpy="0 0 0" xyz="0.3955 0.0016612 0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.6651914291880923"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <link name="tool0"/>
  <joint name="Scraper_Link-tool0" type="fixed">
    <origin rpy="0 0 0" xyz="-0.006 0 0"/>
    <parent link="Scraper_Link"/>
    <child link="tool0"/>
  </joint>
  <joint name="scraper_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="tool0"/>
    <child link="scraper_base_link"/>
  </joint>
  <!-- BASE LINK -->
  <link name="scraper_base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0.026158 -1.2576E-09 -0.00011965"/>
      <mass value="2"/>
      <inertia ixx="5.8586E-05" ixy="-6.3567E-12" ixz="-5.0235E-10" iyy="2.9608E-05" iyz="-8.7386E-08" izz="3.5524E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/lasapc/Documents/Robetarme_ws/robetarme-sfr-v2/t6.1-6.2-sfr-robot/ros2_ws/install/scraper_description/share/scraper_description/meshes/scraper.STL"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.75294 0.75294 0.75294 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/lasapc/Documents/Robetarme_ws/robetarme-sfr-v2/t6.1-6.2-sfr-robot/ros2_ws/install/scraper_description/share/scraper_description/meshes/scraper.STL"/>
      </geometry>
    </collision>
  </link>
</robot>
